Full path in window title option

budRich opened the pull request:
I realized it would be really handy if the full path to the currently open file would be visible in the window title.
I couldn't figure out how to achieve this without modifying core.step(), hence this patch. 

If it is possible to achieve this with a plugin instead, it would be interesting to see how, I am new to Lua and couldn't figure it out.

Also I didn't know how to format the code, i don't think the long line my hack did looks good, but wasn't sure what the best way to do it is.

I also tried formatting the path with, `common.home_encode()`, but figured I actually prefer it being expanded for my usecase, but maybe most people rather have it encoded. Perhaps, making the option a 3 state thing?  
`config.full_path_in_window_title = expanded | yes | no`

budRich commented:
hmm. It seems like it crashes when i open the log (clicking the icon in bottom left) and this is enabled...

franko commented:
Hi,

I like this proposition and I am ok to go with that. Just two things:

- I don't like to have an option for that, just let's do it right away. It is important that we make choices and we leave to the user only the interesting options not the irrelevant details.
- too much logic in the core.step, what about setting a variable in "core" like for "active_view" and just use it in the core.step ? In this way the logic will be moved elsewhere and core.step would be simpler.

budRich commented:
yeah, I added the option, so that it wouldn't be a breaking feature, but I am all for less options. I didn't really think about it, but it is exactly this kind of stuff that makes editor configs such a pain to work with.

This immediateUI/fram refresh workflow is completely new to me, and I was worried that small tests like active_view could be problematic. Im not sure what you mean with setting active_view variable, but i moved DocView = require "core.docview" to core.init() instead.

franko commented:
Congratulation for your first contribution to lite-xl and thank you for your help!

franko commented:
Please have a look at [this branch](https://github.com/franko/lite-xl/tree/window-title-fullpath), I just modified your code to:

- print the real full path of the file by using ~ for the HOME directory
- show if the file has modifications with an asterisk

I think you should agree with these modifications but please have a look, I would like to have your advice.
